doc(partner_sdk): Alter Table operation guidelines (#71)
fivetran/fivetran_sdk: development-guide.md

- This operation might be requested for a table that does not exist in the destination. In that case, it should NOT fail, simply ignore the request and return `success = true`.
- `utc_delete_before` has millisecond precision.

#### AlterTable

> - This operation should exercise change of primary key columns, addition of columns, change of data types, etc. 

Reviewer: @fivetran-rishabhghosh Ending with etc. is very vague. Can you provide more specific operations?

> - This operation should exercise change of primary key columns, addition of columns, change of data types, etc. 

Reviewer: ```suggestion
- This operation should perform changes to primary key columns and data types, column additions, (INSERT OTHER)
```

> - However, this operation should not drop any columns even if the `AlterTable` request has a table with a different set of columns. Dropping columns could lead to unexpected customer data loss and is against Fivetran's general approach to data movement.

Reviewer 2: This doesn't say what they should do. i'd also love to link to a doc that describes our policy/what we usually do - ie not dropping columns/deleting data - but I didn't just find a suitable link.

eg 'this operation should not drop any columns even if the 'AlterTabel' request has a table with a different set of columns. Fivetran's approach to data movement is to leave such columns in a customer's warehouse with no additional data populating to provide the customer control and choice on timing for removal. Dropping columns can lead to unexpected customer data loss.

> - This operation should exercise change of primary key columns, addition of columns and change of data types. 
> - However, this operation should not drop any columns even if the AlterTable request has a table with a different set of columns. Dropping columns could lead to unexpected customer data loss and is against Fivetran's general approach to data movement.

Reviewer: ```suggestion
- This operation should be used for changing primary key columns, adding columns, and changing data types. 
- However, this operation should not drop any columns even if the `AlterTable` request has a table with a different set of columns. Dropping columns could lead to unexpected customer data loss and is against Fivetran's general approach to data movement.
```

> - However, this operation should not drop any columns even if the alter table request has a table with a different set of columns. Dropping of columns could lead to unexpected data loss for customers and is against Fivetran's general approach to data movement.

Reviewer: ```suggestion
- However, this operation should not drop any columns even if the AlterTable request has a table with a different set of columns. Dropping columns could lead to unexpected customer data loss and is against Fivetran's general approach to data movement.
```
